lib/Rex/DetectCLI.pm

```diff
@@ -0,0 +1,89 @@
+#
+# (c) Jan Gehring <jan.gehring@gmail.com>
+#
+# vim: set ts=2 sw=2 tw=0:
+# vim: set expandtab:
+
+package Rex::DetectCLI;
+
+require Exporter;
+use base qw(Exporter);
+use vars qw(@EXPORT);
+
+use List::MoreUtils qw'firstidx';
+use Data::Dumper;
+
+use PPI;
+
+@EXPORT = qw(detect_cli);
+
+sub detect_cli {
```

Comment:
yes, i know, i also feel not that good with it but it seems to work pretty good.